numba/tests/test_chrome_trace.py

```diff
@@ -0,0 +1,51 @@
+import os
+import json
+import unittest
+from textwrap import dedent
+from tempfile import TemporaryDirectory
+
+from numba.tests.support import TestCase, run_in_subprocess
+
+
+class TestChromeTraceModule(TestCase):
+    """
+    Test chrome tracing generated file(s).
+    """
+
+    def test_trace_output(self):
+        code = """
+            from numba import njit
+            import numpy as np
+
+            x = np.arange(100).reshape(10, 10)
+
+            @njit
+            def go_fast(a):
+                trace = 0.0
+                for i in range(a.shape[0]):
+                    trace += np.tanh(a[i, i])
+                return a + trace
+
+            go_fast(x)
+        """
+
+        src = dedent(code)
+        with TemporaryDirectory() as tmpdir:
+            path = os.path.join(tmpdir, "test_trace.json")
```

Comment:
Could this just be `tempfile.NamedTemporaryFile`?

Reply:
No, `NamedTemporaryFile` will create and open the file. We want the chrometrace code to do that. The tempdir guarantee a safe empty directory.

Reply:
Ah yes, good point.